packages/eslint-plugin/src/rules/strict-boolean-expressions.ts

```diff
@@ -76,36 +85,63 @@ export default createRule<Options, MessageId>({
       conditionErrorAny:
         'Unexpected any value in conditional. ' +
```

Comment:
Here and throughout, I think you'll want to use the messaging templating syntax.
```suggestion
        'Unexpected any value in {{context}}. ' +
```
and, usage:
```ts
context.report({
    messageId: 'conditionErrorAny',
    data: { context: 'array predicate' /* or 'conditional' */ }
    // etc
});
```

I think that can simplify the changeset a lot?